aip/auth/4110.md

```diff
@@ -165,6 +176,54 @@ digraph d_front_back {
     1. If a quota project is provided explicitly while initiating ADC, override the quota project in the credential with the explicit value. _[END]_
     1. Else if the `GOOGLE_CLOUD_QUOTA_PROJECT` environment variable is set, override the quota project in the credential with this value. _[END]_
 
+### Retrieval and Caching of Workload Access Tokens from the Metadata Server
+
+When retrieving workload access tokens from the metadata server on GCE, GKE, App
+Engine, or other GCP compute platforms, the following recommendations apply.
+Google-provided auth libraries adhere to these recommendations.  If your
+workload directly communicates with the metadata server in order to retrieve
+workload access tokens, you should adhere to these recommendations.  If you do
+not, your workload may suffer from intermittent and difficult-to-debug
+authentication errors.
+
+**Cache the access token in memory:** Your workload should not make a call to
+the metadata server every time you make a call to a GCP API.  Doing so will
+cause your workload to be rate-limited by the metadata server, or GCP IAM.
+Instead, you should maintain an in-memory cache of the access token, and use the
+cached token across all of your outbound requests.
+
+**Use a robust refresh strategy:**  Each time you attempt to use the cached
+access token, check the remaining lifetime of the token.
+* If it is greater than 5 minutes, the token is fresh.  Proceed to use it.
```

Comment:
Some server-less metadata server refreshes tokens 4 minutes before expiration instead of 5 minutes.

https://github.com/googleapis/google-auth-library-java/blob/082509c09e5fd0ce9ca83cc995517ed3b178c58a/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java#L92 chooses 3m45s with some buffer.

      _static final Duration COMPUTE_REFRESH_MARGIN = Duration.ofMinutes(3).plusSeconds(45);_

Reply:
OK, updated the text to use 225 seconds as the cutoff.